central/cve/cluster/datastore/store/postgres/full_store.go

```diff
@@ -0,0 +1,285 @@
+package postgres
+
+import (
+	"context"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	protoTypes "github.com/gogo/protobuf/types"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stackrox/rox/central/cve/cluster/datastore/store"
+	"github.com/stackrox/rox/central/cve/converter/v2"
+	"github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/generated/storage"
+	ops "github.com/stackrox/rox/pkg/metrics"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	pkgSchema "github.com/stackrox/rox/pkg/postgres/schema"
+	"github.com/stackrox/rox/pkg/set"
+)
+
+// NewFullStore augments the generated store with upsert and delete cluster cves functions.
+func NewFullStore(db *pgxpool.Pool) store.Store {
+	return &fullStoreImpl{
+		db:    db,
+		Store: New(db),
+	}
+}
+
+type fullStoreImpl struct {
+	db *pgxpool.Pool
+
+	Store
+}
+
+func (s *fullStoreImpl) DeleteClusterCVEsForCluster(ctx context.Context, clusterID string) error {
+	conn, release, err := s.acquireConn(ctx, ops.RemoveMany, "ClusterCVE")
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(ctx, "DELETE FROM "+pkgSchema.ClusterCveEdgesTableName+" WHERE clusterid == $1", clusterID)
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.Exec(ctx, "DELETE FROM "+pkgSchema.ClusterCvesTableName+" WHERE not exists (select "+pkgSchema.ClusterCveEdgesTableName+".cveid from "+pkgSchema.ClusterCveEdgesTableName+")", clusterID)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *fullStoreImpl) UpsertClusterCVEParts(ctx context.Context, cveType storage.CVE_CVEType, cvePartsArr ...converter.ClusterCVEParts) error {
+	iTime := protoTypes.TimestampNow()
+	conn, release, err := s.acquireConn(ctx, ops.UpdateMany, "ClusterCVE")
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	tx, err := conn.Begin(ctx)
+	if err != nil {
+		return err
+	}
+
+	cves := make([]*storage.ClusterCVE, 0, len(cvePartsArr))
```

Comment:
take this out of the connection acquisition?

Reply:
Good point